tape/src/main/java/com/squareup/tape/InMemoryObjectQueue.java

```diff
@@ -11,36 +14,57 @@
  * @param <T> The type of elements in the queue.
  */
 public final class InMemoryObjectQueue<T> implements ObjectQueue<T> {
-  private final Queue<T> tasks;
+  // LinkedList can be used both as a List and Queue.
+  private final LinkedList<T> entries;
```

Comment:
ArrayDequeue is missing `subList` (from `AbstractList`) for `peek(n)`. `List` is missing `peek()` and `remove()` (closest would be `get(0)` and `remove(0)`)

 Not terribly hard to write it either ourselves if needed though.